supertokens_python/recipe/session/session_functions.py

```diff
@@ -47,8 +50,36 @@ async def create_new_session(
     if access_token_payload is None:
         access_token_payload = {}
 
+    config = recipe_implementation.config
+    app_info = recipe_implementation.app_info
+    top_level_api_domain = app_info.top_level_api_domain
+    top_level_website_domain = app_info.top_level_website_domain
+
+    if (
+        (not disable_anti_csrf)
+        and config.cookie_same_site == "none"
+        and (not config.cookie_secure)
+        and not (
+            (
+                top_level_api_domain == "localhost"
+                or is_an_ip_address(top_level_api_domain)
+            )
+            and (
+                top_level_website_domain == "localhost"
+                or is_an_ip_address(top_level_website_domain)
+            )
+        )
+    ):
+        # We can allow insecure cookie when both website & API domain are localhost or an IP
+        # When either of them is a different domain, API domain needs to have https and a secure cookie to work
+        raise Exception(
+            "Since your API and website domain are different, for sessions to work, please use https on your api_domain and dont set cookie_secure to false."
+        )
```

Comment:
the location of this will change in the node SDK - so should change here as well